new version of fluidsynth2 2.1.1

docoder-prog opened the pull request:
The info file for fluidsynth-2.1.1 to be added to fink.

docoder-prog commented:
> These changes look good.
> Only thing left here is to add fluidsynth2-dev to the list of packages to avoid in the fink-package-precedence (f-p-p) call (line 50).
> 
Happy to hear that. Will do so.
> In the existing fluidsynth1.info file, you'll also need to add fluidsynth2 and fluidsynth2-dev to the appropriate places in the Conflicts, Replaces, and f-p-p sections (and bump the revision).
> 
Surely.
> Lastly, fluidsynth1 is at ancient v1.1.6. Is there any reason to keep it there and not bump it to the last version (1.1.11 in github) from that series?

I will do that and see if i works.

Till the next time.

docoder-prog commented:
Info4: <<
Package: fluidsynth2
Version: 2.1.1
Revision:1
###
Depends: <<
	%N-shlibs (= %v-%r),
	glib2-shlibs (>= 2.22.4-9),
	libgettext8-shlibs (>=0.20.1-1),
	libsndfile1-shlibs (>= 1.0.28-1),
	readline7-shlibs (>= 7.0.p15-2),
	sdl2-shlibs (>= 2.0.10-2),
	ladspa (>= 1.13-1)
<<
###
BuildDepends: <<
	cmake (>=3.16.2-1),
	fink-package-precedence,
	glib2-dev (>=2.22.4-9),
	libgettext8-dev(>=0.20.1-1),
	libsndfile1-dev (>= 1.0.28-1),
	pkgconfig (>= 0.29.2-1),
	readline7 (>= 7.0.p15-2),
	ladspa-dev (>= 1.13-1),
	sdl2 (>= 2.0.10-2)
<<
Conflicts: fluidsynth, fluidsynth1, fluidsynth2
Replaces: fluidsynth, fluidsynth1, fluidsynth2
###
Source: https://github.com/FluidSynth/fluidsynth/archive/v2.1.1.tar.gz
Source-md5: 165902909092c818a24517de6a4f2f83
SourceRename: fluidsynth-%v.tar.gz
PatchScript: <<
 perl -pi -e 's/\${LIB_SUFFIX}//' CMakeLists.txt src/CMakeLists.txt
 <<
CompileScript: <<
 #!/bin/sh -ev
 %p/bin/cmake \
  -Denable-dbus=0 \
  -Denable-jack=0 \
  -Denable-pulseaudio=0 \
  -DCMAKE_INSTALL_PREFIX=%p/lib/%N \
  -DCMAKE_C_FLAGS="-fno-common -fPIC -MD" \
  -DLIB_SUFFIX="" \
  -Denable-framework:BOOL=OFF \
  -DREADLINE_INCLUDE_DIR:PATH=%p/include \
  -DREADLINE_LIBRARIES:FILEPATH=%p/lib/libreadline.dylib \
  .
 /usr/bin/make
 fink-package-precedence --depfile-ext='\.d' --prohibit-bdep=fluidsynth-dev,fluidsynth1-dev, fluidsynth2-dev .
<<
###
InstallScript: <<
#! /bin/sh -ev
make install DESTDIR=%d
mkdir -p %i/lib/pkgconfig
ln -s %p/lib/fluidsynth/lib/pkgconfig/fluidsynth.pc %i/lib/pkgconfig/fluidsynth.pc
<<
SplitOff: <<
 Package: %N-shlibs
 Depends: <<
  glib2-shlibs (>= 2.22.0-1),
  libgettext8-shlibs (>=0.20.1-1),
  libsndfile1-shlibs (>= 1.0.28-1),
  readline7-shlibs (>= 7.0.p15-2)
 <<
 Files: lib/%N/lib/libfluidsynth.*.dylib
 Shlibs: %p/lib/%N/lib/libfluidsynth.2.dylib 2.0.0 %n (>= 2.1.1-1)
 DocFiles: README.md README.cmake.md
<<
SplitOff2: <<
 Package: %N-dev
 Depends: %N-shlibs
 Files: lib/%N/include lib/%N/lib lib/pkgconfig
 DocFiles: README.md README.cmake.md
 BuildDependsOnly: True
 Conflicts: fluidsynth-dev, fluidsynth1-dev, fluidsynth2-dev
 Replaces: fluidsynth-dev, fluidsynth1-dev, fluidsynth2-dev
<<
###
DocFiles: README.md
###
HomePage: http://fluidsynth.org
License: LGPL
###
Description: Software synthesizer with Soundfont 2 specs
DescDetail: <<
Fluidsynth is a free software synthesizer which is currently based on the
SoundFont standard and supports real time MIDI effect controls. It can be
used as a shared library for embedding in other applications, can play MIDI
files and has a command line shell. Many other applications use FluidSynth
for audio synthesis.
<<
DescUsage: <<
To properly link to fluidsynth2, packages that do not use
the standard pkg-config system will need to set some combination
of CFLAGS, LDFLAGS, or similar compiler flags to point to
%p/lib/%N/(include/lib).
<<
DescPackaging: <<
Disabled D-Bus support: It only is used to interact with rtkit / RealtimeKit,
a Linux-only system service. In other words: On Mac OS X, it did nothing
useful at all.
<<
###
Maintainer: Ebrahim Mayat <emayat@users.sourceforge.net>
<<

docoder-prog commented:
I have uploaded the info file for fluidsynth2 as per your comments

docoder-prog commented:
On May 6, 2020, at 11:11 AM, nieder <notifications@github.com> wrote:
> These changes look good.
> Only thing left here is to add fluidsynth2-dev to the list of packages to avoid in the fink-package-precedence (f-p-p) call (line 50).
> 
> In the existing fluidsynth1.info file, you'll also need to add fluidsynth2 and fluidsynth2-dev to the appropriate places in the Conflicts, Replaces, and f-p-p sections (and bump the revision).
> 
> Lastly, fluidsynth1 is at ancient v1.1.6. Is there any reason to keep it there and not bump it to the last version (1.1.11 in github) from that series?
> 
Hello Hanspeter,

Please find attached the info files for both fluidsynth1 and fluidsynth2. (I have not been able to clone my GitHub fork to my computer yet.)

Best Regards,
Ebrahim

nieder commented:
fluidsynth2.info looks fine, but there's no updated fluidsynth1.info.

docoder-prog commented:
> On May 14, 2020, at 11:29 AM, nieder <notifications@github.com> wrote:
> 
> 
> fluidsynth2.info looks fine, but there's no updated fluidsynth1.info.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/fink/fink-distributions/pull/611#issuecomment-628711050>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APLBGCOLLDXRMMZI3NQ6XBLRRQE5VANCNFSM4MRILI6A>.
> 
Hello Hanspeter,

I am sending it to you (again).

Greetings,
Ebrahim

dmacks commented:
Is the pkgconfig symlink correct? Comparing
```-DCMAKE_INSTALL_PREFIX=%p/lib/%N```
and
```ln -s %p/lib/fluidsynth/lib/pkgconfig/fluidsynth.pc %i/lib/pkgconfig/fluidsynth.pc```
should that latter be %p/lib/fluidsynth2/lib/pkgconfig/fluidsynth.pc (or, for more clarity, %p/lib/%N/lib/pkgconfig/fluidsynth.pc)?

However...

It might save users some annoyance with flags if this new libversion switched back to the default prefix instead of using ```-DCMAKE_INSTALL_PREFIX=%p/lib/%N```. There are technical reasons the fluidsynth1 suite needed to do that (see faf4863bc09665e531eba8f038a0afaf032c64fe), but it's non-standard and jumping to a new package-name is an opportunity to drop that game.

docoder-prog commented:
> Is the pkgconfig symlink correct? Comparing
> `-DCMAKE_INSTALL_PREFIX=%p/lib/%N`
> and
> `ln -s %p/lib/fluidsynth/lib/pkgconfig/fluidsynth.pc %i/lib/pkgconfig/fluidsynth.pc`
> should that latter be %p/lib/fluidsynth2/lib/pkgconfig/fluidsynth.pc (or, for more clarity, %p/lib/%N/lib/pkgconfig/fluidsynth.pc)?

Good catch! Daniel. The symlink is incorrect. But, as you point out below, it may just be better to use the default prefix …

> However...
> 
> It might save users some annoyance with flags if this new libversion switched back to the default prefix instead of using the non-standard path via `-DCMAKE_INSTALL_PREFIX=` at all. There are technical reasons the fluidsynth1 suite needed to do that (see [faf4863](https://github.com/fink/fink-distributions/commit/faf4863bc09665e531eba8f038a0afaf032c64fe)), but it's non-standard and jumping to a new package-name is an opportunity to drop that game.

So, would it be better to revert to:

'-DCMAKE_INSTALL_PREFIX=%p'   ?

E.

docoder-prog commented:
> On May 15, 2020, at 12:35 AM, Daniel Macks <notifications@github.com> wrote:
> 
> 
> Is the pkgconfig symlink correct? Comparing
> -DCMAKE_INSTALL_PREFIX=%p/lib/%N
> and
> ln -s %p/lib/fluidsynth/lib/pkgconfig/fluidsynth.pc %i/lib/pkgconfig/fluidsynth.pc
> should that latter be %p/lib/fluidsynth2/lib/pkgconfig/fluidsynth.pc (or, for more clarity, %p/lib/%N/lib/pkgconfig/fluidsynth.pc)?
> 
Quite correct. Let this be option 1. 
> However...
> 
> It might save users some annoyance with flags if this new libversion switched back to the default prefix instead of using -DCMAKE_INSTALL_PREFIX=%p/lib/%N. There are technical reasons the fluidsynth1 suite needed to do that (see faf4863 <https://github.com/fink/fink-distributions/commit/faf4863bc09665e531eba8f038a0afaf032c64fe>), but it's non-standard and jumping to a new package-name is an opportunity to drop that game.
> 

Option 2 does not work. After altering the fluidsynth2.info <http://fluidsynth2.info/> file (see contents of fluidsynth2.patch file below), the following “Permission denied” error was obtained during installation:


> 42c42
> <   -DCMAKE_INSTALL_PREFIX=%p/lib/%N \
> ---
> >   -DCMAKE_INSTALL_PREFIX=%p \
> 57c57
> < ln -s %p/lib/fluidsynth/lib/pkgconfig/fluidsynth.pc %i/lib/pkgconfig/fluidsynth.pc
> ---
> > ln -s %i/lib/pkgconfig/fluidsynth.pc %p/lib/pkgconfig/fluidsynth.pc
> 67,68c67,68
> <  Files: lib/%N/lib/libfluidsynth.*.dylib
> <  Shlibs: %p/lib/%N/lib/libfluidsynth.2.dylib 2.0.0 %n (>= 2.1.1-1)
> ---
> >  Files: lib/libfluidsynth.*.dylib
> >  Shlibs: %p/lib/libfluidsynth.2.dylib 2.0.0 %n (>= 2.1.1-1)
> 74c74
> <  Files: lib/%N/include lib/%N/lib lib/pkgconfig
> ---
> >  Files: include lib lib/pkgconfig


> -- Installing: /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/include/fluidsynth/types.h
> -- Installing: /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/include/fluidsynth/voice.h
> -- Installing: /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/include/fluidsynth/version.h
> -- Installing: /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/include/fluidsynth.h
> -- Installing: /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/share/man/man1/fluidsynth.1
> mkdir -p /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/lib/pkgconfig
> ln -s /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/lib/pkgconfig/fluidsynth.pc /opt/sw/lib/pkgconfig/fluidsynth.pc
> ln: /opt/sw/lib/pkgconfig/fluidsynth.pc: Permission denied
> ### execution of /tmp/fink.8Ia77 failed, exit code 1
> ### execution of /tmp/fink.OaxHT failed, exit code 1
> Removing runtime build-lock...
> Removing build-lock package...
> /opt/sw/bin/dpkg-lockwait -r fink-buildlock-fluidsynth2-2.1.1-1
> (Reading database ... 44545 files and directories currently installed.)
> Removing fink-buildlock-fluidsynth2-2.1.1-1 ...
> Failed: phase installing: fluidsynth2-2.1.1-1 failed

Regards,
E

nieder commented:
The `ln -s` command is trying to write directly to %p during the install phase, so that won't work. What you're probably trying to do is `ln -s %p/lib/pkgconfig/fluidsynth.pc %i/lib/pkgconfig/fluidsynth.pc`. But that's just a symlink pointing to itself once it gets installed after the .deb is built.

Since you're already using `-DCMAKE_INSTALL_PREFIX=%p`, then there's no need to make a symlink for the fluidsynth.pc file.

The link is there in fluidsynth1 because that package hides its files inside %p/lib/%N and it makes things easier for packages that use fluidsynth to find the .pc file if it's directly in %p/lib/pkgconfig (vs hidden in %p/lib/%N/lib/pkgconfig). So we make the convenience symlink. But if fluidsynth2 is going to go straight into %p, then we don't need the symlink because the fluidsynth.pc file will be in %d/%p/lib/pkgconfig already

docoder-prog commented:
> On May 16, 2020, at 2:50 PM, nieder <notifications@github.com> wrote:
> 
> 
> The ln -s command is trying to write directly to %p during the install phase, so that won't work. What you're probably trying to do is ln -s %p/lib/pkgconfig/fluidsynth.pc %i/lib/pkgconfig/fluidsynth.pc. But that's just a symlink pointing to itself once it gets installed after the .deb is built.
> 
> Since you're already using -DCMAKE_INSTALL_PREFIX=%p, then there's no need to make a symlink for the fluidsynth.pc file.
> 
This does not work: deleting the symlink command (with %p as the install prefix) leads to the formation of the bin and share directories only. In other words, the symlink command is required for the formation of the lib and include directories in addition to the bin and share directories. Strange … but true. Including the symlink command results in :

% ls /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/            
bin	include	lib	share
 
but, this ultimately leads to a “Permission denied” issue during installation as indicated previously.

So, it’s back to -DCMAKE_INSTALL_PREFIX=%p/lib/%N as the working option.

nieder commented:
```
$ ls /sw/build.build/root-fluidsynth2-2.1.1-1/sw
bin/     include/ lib/     share/
```
I set `-DCMAKE_INSTALL_PREFIX=%p` and my InstallScript was locally edited to be just this:
```
InstallScript: <<
  make install DESTDIR=%d
<<
```

docoder-prog commented:
> On May 17, 2020, at 6:30 AM, nieder <notifications@github.com> wrote:
> 
> 
> $ ls /sw/build.build/root-fluidsynth2-2.1.1-1/sw
> bin/     include/ lib/     share/
> I set -DCMAKE_INSTALL_PREFIX=%p and my InstallScript was locally edited to be just this:
> 
> InstallScript: <<
>   make install DESTDIR=%d
> <<
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/fink/fink-distributions/pull/611#issuecomment-629775727>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APLBGCNYGTMN4CR54HMLKVTRR64GHANCNFSM4MRILI6A>.
> 
Very Interesting. I altered the InstallScript section as you indicated but I still get an error:

> /usr/bin/install -d -m 755 /opt/sw/src/fink.build/root-fluidsynth2-dev-2.1.1-1/opt/sw/lib
> /bin/mv /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/lib/pkgconfig /opt/sw/src/fink.build/root-fluidsynth2-dev-2.1.1-1/opt/sw/lib/
> mv: /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/lib/pkgconfig: No such file or directory
> ### execution of /bin/mv failed, exit code 1
> ### execution of /tmp/fink.EC1LS failed, exit code 1
> Removing runtime build-lock...
> Removing build-lock package...
> /opt/sw/bin/dpkg-lockwait -r fink-buildlock-fluidsynth2-2.1.1-1
> (Reading database ... 44545 files and directories currently installed.)
> Removing fink-buildlock-fluidsynth2-2.1.1-1 ...
> Failed: phase installing: fluidsynth2-dev-2.1.1-1 failed

and 

> %  ls /opt/sw/src/fink.build/root-fluidsynth2-2.1.1-1/opt/sw/
> bin	share


Two things different on my setup: I am running MacOS Catalina and zsh (which is now the default on Catalina).

>  % sw_vers
> ProductName:	Mac OS X
> ProductVersion:	10.15.4
> BuildVersion:	19E287
> 
>  % echo $SHELL
> /bin/zsh

From your CLI output, it looks like you are running bash on an earlier OS ?

nieder commented:
Can you post your full build log? `fink -l rebuild fluidsynth2` saves the log to /tmp, then zip it and attach it here.

docoder-prog commented:
> On May 17, 2020, at 9:03 AM, nieder <notifications@github.com> wrote:
> 
> 
> Can you post your full build log? fink -l rebuild fluidsynth2 saves the log to /tmp, then zip it and attach it here.
> 
Sure.

nieder commented:
Your `Files:` field in the SplitOff2 package needs to be fixed. I'm assuming it looks something like this?
```
Files: include lib lib/pkconfig
```
This moves the entire "include" directory, then the entire "lib" directory, then "lib/pkgconfig", but that already got moved when moving "lib".

docoder-prog commented:
> On May 19, 2020, at 4:06 PM, nieder <notifications@github.com> wrote:
> 
> 
> Your Files: field in the SplitOff2 package needs to be fixed. I'm assuming it looks something like this?
> 
> Files: include lib lib/pkconfig
> This moves the entire "include" directory, then the entire "lib" directory, then "lib/pkgconfig", but that already got moved when moving "lib”.
> 
nieder,

Thanks for the useful information. “lib/pkgconfig” is indeed redundant. Once this redundancy has been edited out from the Files field, everything compiles and installs without error. The updated fluidsynth2.info file has been added to the master branch on decoder-prog/fink-distributions:

% git add fluidsynth2.info           

% git status
On branch master
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   fluidsynth2.info

 % git commit
[master 80b9eb42ab] fluidsynth2.info has been edited: the Files field in the Splitoff2 package has been edited so that it compiles and installs without error.
 1 file changed, 4 insertions(+), 6 deletions(-)
 
Please revise and merge.

dmacks commented:
fluidsynth1 needs to remain in the same special prefix as it always was. It was like that for a reason, and that reason is still true. Also, by fink policy, one mus never change the pathname in the Shlibs once such a package exists ("fluidsynth1-shlibs" must remain "%p/lib/%N/lib/libfluidsynth.1.dylib" for all eternity). Creating the *new* fluidsynth2 suite is the chance to change that, such as obviously going from libfluidsynth1.dylib to libfluidsynth.2.dylib, but also switching back up to %p/lib

docoder-prog commented:
> On May 21, 2020, at 1:05 AM, Daniel Macks <notifications@github.com> wrote:
> 
> 
> fluidsynth1 needs to remain in the same special prefix as it always was. It was like that for a reason, and that reason is still true. Also, by fink policy, one mus never change the pathname in the Shlibs once such a package exists ("fluidsynth1-shlibs" must remain "%p/lib/%N/lib/libfluidsynth.1.dylib" for all eternity). Creating the new fluidsynth2 suite is the chance to change that, such as obviously going from libfluidsynth1.dylib to libfluidsynth.2.dylib, but also switching back up to %p/lib
> 
Okay, so I’ll revert the fluidsynth1.info <http://fluidsynth1.info/> file back to the Install prefix %p/lib/%N.

I also wanted to add a note on standalone functionality. Running the “fluidsynth” binary normally as in:
% fluidsynth —options files 

leads to an error in Catalina. 

fluidsynth: error: Failed to open the default audio device. Status=-50

This binary has to be run with sudo privilege as in

% sudo /opt/sw/lib/fluidsynth1/bin/fluidsynth -g 2 -a coreaudio -m coremidi  <sf2 file> <MIDI file>

Does this glitch occur on earlier OSes ?   So, a short note on this glitch would be appropriate.

nieder commented:
I don't need sudo on 10.13.
```
nieder $ /sw/lib/fluidsynth1/bin/fluidsynth -g 2 -a coreaudio -m coremidi Piano\ Bass.sf2 Queen\ -\ Bohemian\ Rhapsody.mid 
FluidSynth version 1.1.6
<and plays music>
```
Also works with the executable from fluidsynth2.

docoder-prog commented:
> On May 21, 2020, at 8:26 AM, nieder <notifications@github.com> wrote:
> 
> 
> I don't need sudo on 10.13.
> 
> nieder $ /sw/lib/fluidsynth1/bin/fluidsynth -g 2 -a coreaudio -m coremidi Piano\ Bass.sf2 Queen\ -\ Bohemian\ Rhapsody.mid 
> FluidSynth version 1.1.6
> <and plays music>
> Also works with the executable from fluidsynth2.
> 
Thanks for confirming, nieder. So, if everything is okay, we could close this ticket ?

docoder-prog commented:
> On May 22, 2020, at 8:04 AM, nieder <notifications@github.com> wrote:
> 
> 
> @nieder requested changes on this pull request.
> 
> Reiterating @dmacks <https://github.com/dmacks> comment:
> 
> These fluidsynth1.info and fluidsynth2.info are in the top-level dist directory, not down in ..../sound/
> I highly recommend reopening with a fresh pull request that works off a fluidsynth specific branch from your fork of fink-distributions, rather than putting your changes directly into the decoder-prog:master branch. 
> 
nieder

OK, I created a new branch “fluidsynth” and added the 2 info files 

% git checkout -b fluidsynth
Switched to a new branch 'fluidsynth'

 % git branch
* fluidsynth
  master
  remotes/origin/docoder-prog
  remotes/upstream/docoder-prog

% git add fluidsynth1.info
% git add fluidsynth2.info <http://fluidsynth2.info/>

but …

 % git ls-tree --name-only fluidsynth
.gitignore
10.2-gcc3.3
10.2
10.3
10.4-transitional
10.4
10.7
10.9-libcxx
README
VERSION
compare-trees.pl
fluidsynth1.info
fluidsynth2.info
inject.pl
stamp-cvs-live
 
> The history from this PR now has a bunch of merge commits and pulled in 25 other files that are not relevant. And it's going to push fluidsynth*.info into the top level directory instead of the sound subdirectory.
> 
It looks like even the fluidsynth branch has a bunch of other files in addition to the fluidsynth info files. Would it be wise to push then from the fluidsynth branch as it is now ?

docoder-prog commented:
> The history from this PR now has a bunch of merge commits and pulled in 25 other files that are not relevant. And it's going to push fluidsynth*.info into the top level directory instead of the sound subdirectory.
> 
OK, so the new info files were moved into the directory

fink-distributions/10.9-libcxx/stable/main/finkinfo/sound/

% git push --set-upstream origin fluidsynth  //branch pushed to remote copy of clone
Total 0 (delta 0), reused 0 (delta 0)
remote: 
remote: Create a pull request for 'fluidsynth' on GitHub by visiting:
remote:      https://github.com/docoder-prog/fink-distributions/pull/new/fluidsynth
remote: 
To https://github.com/docoder-prog/fink-distributions
 * [new branch]            fluidsynth -> fluidsynth
Branch 'fluidsynth' set up to track remote branch 'fluidsynth' from 'origin’.

 % git commit
On branch fluidsynth
Your branch is up to date with 'origin/fluidsynth'.

Changes not staged for commit:
	modified:   10.9-libcxx/stable/main/finkinfo/sound/fluidsynth1.info
	deleted:    fluidsynth1.info
	deleted:    fluidsynth2.info

Untracked files:
	10.9-libcxx/stable/main/finkinfo/sound/fluidsynth2.info

no changes added to commit

 % git push
Everything up-to-date

% git pull
Already up to date.

docoder-prog commented:
> Reiterating @dmacks comment:
> 
> ```
> These fluidsynth1.info and fluidsynth2.info are in the top-level dist directory, not down in ..../sound/
> ```
> 
> I highly recommend reopening with a fresh pull request that works off a fluidsynth specific branch from your fork of fink-distributions, rather than putting your changes directly into the decoder-prog:master branch. The history from this PR now has a bunch of merge commits and pulled in 25 other files that are not relevant. And it's going to push fluidsynth*.info into the top level directory instead of the sound subdirectory.
> 
> https://github.com/fink/fink-distributions/wiki/Submitting-new-or-updated-packages

That sounds good to me. I have opened a fresh pull request called "fluidsynth" with the fluidsynth info files in the fink-distributions/10.9-libcxx/stable/main/finkinfo/sound directory.

I am therefore closing this pull request.
